### DISCUSSION TITLE
How to mock useState hook and check if setState have been called with vitest?

### DISCUSSION THREAD

**Discussion by siyi98** [vitest-dev/vitest#3699] [Q&A]
I'm testing a component which must call the setState function in certain situations and with jest I did something like:
```ts
 it('Component should call setState on render', () => {
        const setState = jest.fn();

        jest.spyOn(React, 'useState').mockReturnValue(['', setState]);

        render(<Component />);

        expect(setState).toHaveBeenCalledWith('text');
    });
```

However, when using vitest - and properly changing 'jest.' for 'vi.' - this test does no work and I receive the following message:

```ts
 FAIL  src/__tests__/components/Component.spec.tsx > Component > Component should call setState on render
AssertionError: expected "spy" to be called with arguments: [ 'text' ]expected "spy" to be called with arguments: [ 'text' ]

Received:


Number of calls: 0

 ❯ src/__tests__/components/Component.spec.tsx:82:19
     80|   );
     81|
     82|   expect(setState).toHaveBeenCalledWith('text');
       |                   ^
     83|  });
     84|
```
How this implementation should be made with vitest? Its mocking function appears to works differently from jest.

**Comment by SvetaSR** [comment]
Were you able to solve it?

**Comment by ntrp** [comment]
I am having the same issue, any news?

**Comment by FamALouiz** [comment]
If you could provide a snippet of your source code, that would be amazing. However, here was what my problem was. 

When I was using useState, I was importing it individually as so:
```js
import { useState } from "react"

export default function Component() {

      const [state, setState] = useState(false);  

      return <></>;
}

```
After I converted to using `React.useState`, it worked normally. The modified code would look something like:

```js
import React from "react"

export default function Component() {

     const [state, setState] = React.useState(false);  // Note the React. before the useState

     return <></>;
}
```
I have not yet looked into why that is, but it worked. Try this out.
